src/Workspaces/CSharp/Portable/Extensions/ExpressionSyntaxExtensions.cs

```diff
@@ -2271,14 +2274,21 @@ private static bool CanReplaceWithReducedName(this NameSyntax name, TypeSyntax r
                 IsAmbiguousCast(name, reducedName) ||
                 IsNullableTypeInPointerExpression(name, reducedName) ||
                 name.IsNotNullableReplaceable(reducedName) ||
-                IsQualifiedNameInUsingDirective(semanticModel, name, reducedName))
+                (IsQualifiedNameInUsingDirective(semanticModel, name, reducedName) && !IsGlobalAliasQualifiedName(name)))
```

Comment:
can you extract a named method for this?  to keep all the cases just simple || checks with clear names?

Reply:
Moved these changes into https://github.com/dotnet/roslyn/pull/34964